vm/chef/cookbooks/mariadb/files/mariadb-utils

```diff
@@ -0,0 +1,323 @@
+#!/bin/bash
+
+source /opt/c2d/c2d-utils || exit 1
+
+declare -r replication_user="replication"
+declare -r backup_folder="/tmp/transfer"
+declare -r backup_filename="${backup_folder}/all-databases.sql"
+
+#######################################
+# Prints a formatted message to standard output.
+# Arguments:
+#  Message string.
+#######################################
+function emit_msg() {
+  echo "$(date +"%I:%M:%S")" "${@}"
+}
+
+#######################################
+# Run a command on local MariaDB server using root user.
+# Arguments:
+#  A MariaDB/MySQL command.
+#######################################
+function run_query() {
+  local -r command="$1"
+  mysql --user=root -e "${command}"
+}
+
+#######################################
+# Add short hostnames to /etc/hosts in order to
+# avoid error using long hostnames.
+# Arguments:
+#  Message string.
+#######################################
+function add_short_hostnames() {
+  local -r nodes="$1"
+  local target_ip=""
+  local target_host=""
+
+  for node in ${nodes}; do
+    # target_ip="$(host ${node} | awk -F 'has address' '{ print $2 }' | tr -d ' ')"
+    target_ip="$(getent hosts ${node} | awk '{ print $1 }')"
+    target_host="$(echo ${node} | grep -o -P "mariadb-vm-(\d*)")"
+    echo "${target_ip} ${target_host}" >> /etc/hosts
+  done
+}
+
+#######################################
+# Configure MariaDB to bind service to all available addresses
+#######################################
+function bind_to_all_addresses() {
+  sed -i "s/.*bind-address.*/bind-address = 0.0.0.0/" \
```

Comment:
why do we need to sed it? can be set in config that you're patching

Reply:
Done.

Reply:
Rolling back changes due different importing order on mariadb.conf and conf.d folders. 

```
# The MariaDB/MySQL tools read configuration files in the following order:
# 0. "/etc/mysql/my.cnf" symlinks to this file, reason why all the rest is read.
# 1. "/etc/mysql/mariadb.cnf" (this file) to set global defaults,
# 2. "/etc/mysql/conf.d/*.cnf" to set global options.
# 3. "/etc/mysql/mariadb.conf.d/*.cnf" to set MariaDB-only options.
# 4. "~/.my.cnf" to set user-specific options.
# ...

Result: 

```
root@mariadb-vm-0:/etc/mysql/conf.d# cat replication.cnf 
[mysqld]
server-id=1
log_bin=/var/log/mysql/mysql-bin.log
expire_logs_days=10
max_binlog_size=1000M
binlog-format=row
log-basename=master
binlog-do-db=example_database
bind-address=0.0.0.0
root@mariadb-vm-0:/etc/mysql/conf.d# mysql -e "show variables" | grep bind
bind_address	127.0.0.1
```

Not binding to 0.0.0.0